content/concepts/module-resolution.md

```diff
@@ -3,70 +3,73 @@ title: Module Resolution
+You can replace the original module path by an alternate path by creating an alias for it using [`resolve.alias`](/configuration/resolve/#resolve-alias) configuration option.
+
+Once the path is resolved based on the above rule, the resolver checks to see if the path points to a file or a directory. If the path points to a file:
+* If the path has a file extension, then the file is bundled straightaway.
+* Otherwise, the file extension is resolved using the [`resolve.extensions`](/configuration/resolve/#resolve-extensions) option, which tells the resolver which extensions (eg - `.js`, `.jsx`) are acceptable for resolution.
 
-Once the path is resolved based on the above rule, the resolver checks if the path points to a file or to a directory. If the path points to a file then it is bundled straightaway.
-But if the path points to a folder, then the following steps are taken to find the right file with the right extension.
-* The right file is determined using the `main: "<filename>.js"` field in `package.json`.
-* If there is no package.json or the main file is not found, `resolve.mainFiles` configuration option is looked up.
-* If this also fails, then it looks for a file named `index` by default.
-* `resolve.extensions` tells the resolver which extensions (eg - `.js, .jsx`) are acceptable for resolution.
+If the path points to a folder, then the following steps are taken to find the right file with the right extension:
+* If the folder contains a `package.json` file, then fields specified in [`resolve.mainFiles`](/configuration/resolve/#resolve-mainfields) configuration option are looked up in order, and the first such field in `package.json` determines the file path. 
+* If there is no `package.json` or if the main fields do not return a valid path, file names specified in the [`resolve.mainFiles`](/configuration/resolve/#resolve-mainfiles) configuration option are looked for in order, to see if a matching filename exists in the imported/required directory .
+* The file extension is then resolved in a similar way using the `resolve.extensions` option.
+
+Webpack provides reasonable [defaults](/configuration/resolve) for these options depending on your build target.
 
 ## Resolving Loaders
 
-This follows the same rules as the file resolver. But `resolveLoader` configuration can be used to have separate resolution rules for loaders.
+This follows the same rules as those specified for file resolution. But the [`resolveLoader`](/configuration/resolve/#resolveloader) configuration option can be used to have separate resolution rules for loaders.
 
 ## Caching
 
-Every filesystem access is cached so that multiple parallel or serial requests to the same thing are merged. In watching mode only changed files are removed from cache (the watcher knows which files got changed). In non-watching mode the cache is purged before every compilation.
-
-### Unsafe caching
```

Comment:
Removing the section on unsafe caching since, it's hardly ever used in practice and from what I could make out, webpack doesn't recommend using it. Also it might be suited for a more nice audience (for benchmarking webpack builds etc.) We could add it back if reviewers feel otherwise.